--- a/drivers/gpu/drm/i915/intel_runtime_pm.c
+++ b/drivers/gpu/drm/i915/intel_runtime_pm.c
@@ -32,6 +32,19 @@
 #include "i915_drv.h"
 #include "intel_drv.h"
 
+int pm_runtime_get_if_in_use(struct device *dev)
+{
+       unsigned long flags;
+       int retval;
+
+       spin_lock_irqsave(&dev->power.lock, flags);
+       retval = dev->power.disable_depth > 0 ? -EINVAL :
+               dev->power.runtime_status == RPM_ACTIVE
+                       && atomic_inc_not_zero(&dev->power.usage_count);
+       spin_unlock_irqrestore(&dev->power.lock, flags);
+       return retval;
+}
+
 /**
  * DOC: runtime pm
  *
