diff --git a/drivers/gpu/drm/drm_prime.c b/drivers/gpu/drm/drm_prime.c
index 7482b06..bf2523e 100644
--- a/drivers/gpu/drm/drm_prime.c
+++ b/drivers/gpu/drm/drm_prime.c
@@ -33,6 +33,7 @@
 
 #include "drm_internal.h"
 
+#include "backport.h"
 /*
  * DMA-BUF/GEM Object references and lifetime overview:
  *
diff --git a/include/backport.h b/include/backport.h
index 2a22e8b..ab30b47 100644
--- a/include/backport.h
+++ b/include/backport.h
@@ -24,3 +24,7 @@ static inline void __iomem *acpi_os_ioremap(acpi_physical_address phys,
        return ioremap_cache(phys, size);
 }
 
+
+#undef dma_buf_export
+#define dma_buf_export(priv, ops, size, flags, resv)	\
+	dma_buf_export_named(priv, ops, size, flags, KBUILD_MODNAME)
