--- a/drivers/gpu/drm/i915/Makefile
+++ b/drivers/gpu/drm/i915/Makefile
@@ -181,10 +181,10 @@ i915-y += i915_perf.o \
 	  i915_oa_cnl.o \
 	  i915_oa_icl.o
 
-ifeq ($(CONFIG_DRM_I915_GVT),y)
-i915-y += intel_gvt.o
-include $(src)/gvt/Makefile
-endif
+#ifeq ($(CONFIG_DRM_I915_GVT),y)
+#i915-y += intel_gvt.o
+#include $(src)/gvt/Makefile
+#endif
 
 # LPE Audio for VLV and CHT
 i915-y += intel_lpe_audio.o
