--- a/drivers/gpu/drm/i915/i915_sw_fence.c
+++ b/drivers/gpu/drm/i915/i915_sw_fence.c
@@ -14,6 +14,25 @@
 
 #include "i915_sw_fence.h"
 
+static inline struct fence *
+reservation_object_get_excl_rcu(struct reservation_object *obj)
+{
+        struct fence *fence;
+        unsigned seq;
+retry:
+        seq = read_seqcount_begin(&obj->seq);
+        rcu_read_lock();
+        fence = rcu_dereference(obj->fence_excl);
+        if (read_seqcount_retry(&obj->seq, seq)) {
+                rcu_read_unlock();
+                goto retry;
+        }
+        fence = fence_get(fence);
+        rcu_read_unlock();
+        return fence;
+}
+
+
 static DEFINE_SPINLOCK(i915_sw_fence_lock);
 
 static int __i915_sw_fence_notify(struct i915_sw_fence *fence,
